user-defined-plugin.md

```diff
@@ -0,0 +1,261 @@
+These two files need to be stored inside `~/.exploratory/plugins` directory as follows.
+
+`~/.exploratory/plugins/<plugin_name>/plugin.json`
+
+`~/.exploratory/plugins/<plugin_name>/scripts/plugin.r`
+ 
+
+## Create Plugin directory
+
+So let's say you want to create a plugin that uses `riem_measures` function from [riem](https://github.com/ropensci/riem) R package to get ASOS data via the Iowa Environment Mesonet. Then first you want to to create a directory called `riem_measures`. NOTE: The directory name is the plugin name and should be unique among user-defined plugins.
+
+So on Mac, you can do:
+
+`$ cd ~/.exploratory/plugins`
+
+`$ mkdir riem_measures`
+
+## Create Plugin Definition Meta File.
+
+This is how your Plugin Definition Meta File (`plugin.json`) looks like. The file must be in valid JSON format.
+
+```
+{
+  "name": "riem_measures",
+  "displayName": "riem measures",
+  "iconURL" : "lib/images/plugin.png",
+  "helpURL" :  "http://ropensci.github.io/riem/",
+  "rPackageDependencies": ["riem"],
+  "function": "riem_measures",
+  "rSourceFile" : "plugin.r",
+  "hasQueryField" : false,
+  "inputParameters": [
+    {
+      "name": "station",
+      "displayName":"Station",
+      "colSpan" : 5,
+      "dataType": "text",
+      "defaultValue" : "",
+      "required" : true
+    },
+    {
+      "name": "date_start",
+      "displayName":"Date Start",
+      "colSpan" : 5,
+      "dataType": "text",
+      "defaultValue" : "",
+      "required" : true
+    },
+    {
+      "name": "date_end",
+      "displayName":"Date End",
+      "colSpan" : 5,
+      "dataType": "text",
+      "defaultValue" : "",
+      "required" : true
+    }
+  ]
+}
+```
+
+### Attributes
+
+#### name (required)
+
+`name` attribute holds Name of the plugin. in this case `riem_measurs`, please make sure to use your directory name for this `name` attribute. 
+
+#### displayName (required)
+
+`displayName` attribute is used for Display Name on Data Source Picker Dialog and Import Data Dialog.
+
+#### iconURL (required)
+
+`iconURL` holds icon image file URL for your plugin. currently it only supports external URL.
+
+For example:
+ 
+```
+   "iconURL" : "http://xmllondon.com/images/sparqlThumb.png",
+```
+
+If you do not have an icon, please use below for default lego like block icon.
+
+```
+   "iconURL" : "lib/images/plugin.png",
+```
+
+If you use external URL, make sure to set
+`iconWidh` and `iconHeight` to 32px to fit icon in the UI.
+
+```
+"iconWidth" : "32px",
+"iconHeight" : "32px"
+```
+
+#### helpURL (required)
+
+`helpURL` holds URL for your plugin help page. Help Link is put on Import Dialog Header.
+if you do not have one, you can set default exploratory doc link like below:
+
+```
+"helpURL" :  "http://docs.exploratory.io/",
+```
+
+#### rPackageDependencies (required)
+
+`rPackageDependencies` is an array of package names that the plugin depends. If your plugin does not depends on any package,
+leave it as empty array like below:
```

Comment:
fixed it. Actually, it's already automatically set empty array. So changed this attribute to optional.